### PR TITLE
fix pshm_large_than_4GB_windows

### LINKED ISSUE #106: Unable create share memory more than 4GB on windows
Hi~ When i use plibsys in my program, i foud it can't create a share memory which size is more than 4GB on windows.  In the “pshm-win.c”, line 74. function CreateFileMappingA(), argument dwMaximumSizeHigh was set to 0. And that cause this problem. 
Could you fix this?
Thanks.
Best.

### PR DESCRIPTION
**Pull request by leekunpeng**
fix #106

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 CMakeLists.txt.user
 .DS_Store
+build
```
Comment (saprykin) [inline .gitignore:3]: Can you please revert this? We do not want to add personalised build-related settings.

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "files.associations": {
+        "*.out": "yaml",
+        "pmem.h": "c",
+        "psemaphore.h": "c"
+    }
+}
```
Comment (saprykin) [inline .vscode/settings.json:7]: The same here, this file needs to be removed.

File: tests/pshm_test.cpp [hunk 3]
```diff
@@ -305,11 +305,37 @@ P_TEST_CASE_BEGIN (pshm_thread_test)
...
+}
+P_TEST_CASE_END ()
+
 P_TEST_SUITE_BEGIN()
 {
 	P_TEST_SUITE_RUN_CASE (pshm_nomem_test);
 	P_TEST_SUITE_RUN_CASE (pshm_invalid_test);
 	P_TEST_SUITE_RUN_CASE (pshm_general_test);
 	P_TEST_SUITE_RUN_CASE (pshm_thread_test);
+	P_TEST_SUITE_RUN_CASE (pshm_lagrge_4GB_win_test);
```
Comment (saprykin) [inline tests/pshm_test.cpp:339]: This has to be removed/reverted back. You did it correctly removing in the previous commit, I do not know why you have decided to bring it back again.

**Review by saprykin — CHANGES_REQUESTED** [review]
Hi, I reviewed the changes which should be easy to address. I could have quickly fixed them as well, but I do not have permissions to push to your forked repo (see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

File: .gitignore (hunk 1 above)
Comment (leekunpeng) [reply .gitignore]: sorry for that, i have just delete it

File: .vscode/settings.json (hunk 2 above)
Comment (leekunpeng) [reply .vscode/settings.json]: okay, i delete it

File: tests/pshm_test.cpp (hunk 3 above)
Comment (leekunpeng) [reply tests/pshm_test.cpp]: Sorry for that, i is my first time to use PR. I must make mistakes.
I have already delete it.

**Review by saprykin — APPROVED** [review]
Thanks, looks goof for merge.

**Pull request merged**